internal/types/types.go

```diff
@@ -188,3 +188,10 @@ type UserInfoService interface {
 	// and unpacks it into the UserInfo type.
 	FetchUserInfoFromIssuer(ctx context.Context, iss, rawToken string) (*UserInfo, error)
 }
+
+// OAuthClientManager defines the storage interface for OAuth clients.
+type OAuthClientManager interface {
+	CreateOAuthClient(ctx context.Context, client OAuthClient) (OAuthClient, error)
```

Comment:
Why do we return an `OAuthClient` here instead of a `*OAuthClient`?

Reply:
This is just a struct of data, with no methods (apart from the type conversion), so I thought it'd be better if these return the structs directly instead of pointers.